### ISSUE TITLE
Uncaught TypeError: Cannot read property 'marks' of undefined

### ISSUE THREAD

**Issue by chemzqm** [ianstormtaylor/slate#1285]
#### Do you want to request a *feature* or report a *bug*?

Bug of slate 0.27.4 with react 16

#### What's the current behavior?

Previous text node could be empty, so it will throws and the entire editor get crashed.

#### What's the expected behavior?

Should never crash.

The solution is:
``` js
diff --git a/packages/slate/src/models/node.js b/packages/slate/src/models/node.js
index 43d93d8a..549b37a0 100644
--- a/packages/slate/src/models/node.js
+++ b/packages/slate/src/models/node.js
@@ -1057,14 +1057,14 @@ class Node {
       const previous = this.getPreviousText(startKey)
       if (!previous || previous.text.length == 0) return []
       const char = previous.characters.get(previous.text.length - 1)
-      return char.marks.toArray()
+      return char ? char.marks.toArray() : []
     }
 
     // If the range is collapsed, check the character before the start.
     if (range.isCollapsed) {
       const text = this.getDescendant(startKey)
       const char = text.characters.get(range.startOffset - 1)
-      return char.marks.toArray()
+      return char ? char.marks.toArray() : []
     }
 
     // Otherwise, get a set of the marks for each character in the range.
@@ -1094,14 +1094,14 @@ class Node {
       const previous = this.getPreviousText(startKey)
       if (!previous || !previous.length) return []
       const char = previous.characters.get(previous.length - 1)
-      return char.marks.toArray()
+      return char ? char.marks.toArray() : []
     }
 
     // If the range is collapsed, check the character before the start.
     if (range.isCollapsed) {
       const text = this.getDescendant(startKey)
       const char = text.characters.get(range.startOffset - 1)
-      return char.marks.toArray()
+      return char ? char.marks.toArray() : []
     }
 
     // Otherwise, get a set of the marks for each character in the range.

```

**Comment by YurkaninRyan** [comment]
I bet this was fixed in 0.27.5 by #1268 

Can you try that out @chemzqm ?

**Comment by chemzqm** [comment]
@YurkaninRyan It's not, I tried with latest master branch, same problem exists

![](https://gfycat.com/LateClutteredGentoopenguin)

This error happens when I move the cursor to the end of an inline tag which positioned at the end of a block tag.
